static/js/reactions.js

```diff
@@ -13,8 +13,26 @@ exports.open_reactions_popover = function () {
     return true;
 };
 
+function should_send_reaction(emoji_name, operation) {
+    // If a default emoji with this name exists then always send it irrespective
+    // of whether a realm emoji with this name exists or not.
+    if (!emoji_codes.name_to_codepoint.hasOwnProperty(emoji_name)) {
+        // While adding a reaction and a default emoji with this name doesn't
+        // exist then send the request only if there is an active realm emoji
+        // with this name.
+        if (operation === "add") {
+            return emoji.active_realm_emojis.hasOwnProperty(emoji_name);
+        }
```

Comment:
Maybe add a comment in this case mentioning that this behavior isn't correct, in that you can't +1 a deactivated realm emoji, but that more work is required to support that correctly.